crates/solver/src/solver.rs

```diff
@@ -83,6 +87,35 @@ pub trait Solver: Send + Sync + 'static {
     fn name(&self) -> &str;
 }
 
+/// The result a given solver achieved in the auction
+pub enum AuctionResult {
+    /// Solution was valid and was ranked at the given place
+    /// Rank 1 means the solver won the competition
+    Ranked(usize),
+
+    /// Solution was invalid for some reason
+    Rejected(SolverRejectionReason),
+}
+
+/// Reason for why a solution may have been invalid
+pub enum SolverRejectionReason {
+    /// The solver didn't return a successful response in time
+    RunError(SolverRunError),
```

Comment:
Comment says timeout but the actual type differentiates timeout and generic error.

Reply:
Fixed.